### ISSUE TITLE
Investigate "r.next is not a function" console error in conduit-rxjs-umd-min.js

### ISSUE THREAD

**Issue by james-anderson-iu** [indiana-university/conduit#106]
This seems to be logged by RxJS through conduit-rxjs code calling `.next` when a stream executing an AJAX request has a side effect that  redirects the browser to a new page. This does not seem to affect the application since the error is happening in code on a page that is about to be replaced. I would just like to know if there is any way to keep the error from being thrown into the console to avoid adding useless error messages to the console. And if there is a way, can we implement it in conduit-rxjs?

**Comment by basham** [comment]
James and I investigated this a little further today. The error looks to be coming from [line 36 in the run function](https://github.com/indiana-university/conduit/blob/conduit-rxjs%400.6.0/packages/conduit-rxjs/src/run.js#L36).

```js
.forEach(({ subject$, nextValue }) => subject$.next(nextValue))
```

To fix it, we can likely just update the [previous line](https://github.com/indiana-university/conduit/blob/conduit-rxjs%400.6.0/packages/conduit-rxjs/src/run.js#L35), so the filter function explicitly only allows observables.

```js
.filter(({ subject$ }) => isObservable(subject$))
```

**Comment by james-anderson-iu** [comment]
Adding relevant parts of slack conversation. Will convert issue type and PR changes discussed.

_james-anderson-iu_: Well, the `isObservable` check didn't stop the error from being thrown in Conduit. I tried looking at the actual subject right before the `subject$.next()` and didn't notice any reason it couldn't call `next()`. I did find something that seems to stop the error from being thrown. I tried changing the `subject$.next(nextValue)` to `subject$.next?.(nextValue)` so it won't call `next()` if it doesn't exist.

_basham_: For reference, here is the `isObservable` source code.
https://github.com/ReactiveX/rxjs/blob/7.5.5/src/internal/util/isObservable.ts
It turns out, Observables do not have a `next()` method. Subjects extend Observables and those have a `next()` method.
https://github.com/ReactiveX/rxjs/blob/7.5.5/src/internal/Subject.ts#L60
There is no official `isSubject()` utility.
I recommend having the `isObservable` check in addition to the `next?` check. That will mean that it has to be an Observable (or extension of it) and have the `next()` method. It will purposefully exclude generic objects that just happen to have a `next` property.

_james-anderson-iu_: Does it make sense to add the same checks in `connect.js` in `conduit-rxjs-react`, even though we've not had any issue with that code?
`next()` is also used in `createHandler.js`. There is already an observable check there, but we could at the `?.` on the `next()` call there too. Then, between the three changes, all the places that call `next()` would have the same checks in place.

_basham_: That’s fine with me.
 It makes sense to just put the same protections in place everywhere.
